README.md

```diff
@@ -0,0 +1,222 @@
+  useCase: 'useCase',
+});
+```
+
+### Initializing SuperfaceTest instance
+
+```typescript
+import { SuperfaceTest } from '@superfaceai/testing-lib';
+```
+
+without any arguments:
+
+```typescript
+const superface = new SuperfaceTest();
+```
+
+with superface configuration:
+
+```typescript
+const client = new SuperfaceClient();
+const profile = await client.getProfile('profile');
+const provider = await client.getProvider('provider');
+const useCase = await profile.getUseCase('useCase');
+
+const superface = new SuperfaceTest({
+  client,
+  profile,
+  provider,
+  useCase,
+});
+```
+
+with superface and nock configuration:
+
+```typescript
+const superface = new SuperfaceTest(
+  {
+    profile: 'profile',
+    provider: 'provider',
+    useCase: 'useCase',
+  },
+  {
+    path: 'nock-recordings',
+    fixture: 'my-recording',
+    enableReqheadersRecording: true,
+  }
+);
+```
+
+Given superface configuration is stored in class and used later in `run()` method. You can also pass in instance of `SuperfaceClient`, but it is not required as it gets initialized inside library if not provided.
+
+Given nock configuration is also stored in class. Property `path` and `fixture` is used to configure location of recordings and property `enableReqheadersRecording` is used to enable/disable recording of request headers (This is turned off by default).
+
+### Running
+
+To test your capabilities, use method `run()`, which encapsulates nock recording and usecase perform. It expects superface configuration (similar to initializing `SuperfaceTest` class) and input. You don't need to specify profile, provider or useCase if you already specified them when initializing `SuperfaceTest` class.
+
+```typescript
+import { SuperfaceTest } from '@superfaceai/testing-lib';
+
+const superface = new SuperfaceTest({
+  profile: 'profile',
+  provider: 'provider',
+  useCase: 'useCase',
+});
+
+superface.run({
+  input: {
+    some: 'input',
+  },
+});
+```
+
+Example with jest:
+
+```typescript
+import { SuperfaceTest } from '@superfaceai/testing-lib';
+
+describe('test', () => {
+  let superface: SuperfaceTest;
+
+  afterEach(() => {
+    superface = new SuperfaceTest();
+  });
+
+  it('performs corretly', async () => {
+    await expect(
+      superface.run({
+        profile: 'profile',
+        provider: 'provider',
+        useCase: 'useCase',
+        input: {
+          some: 'input',
+        },
+      })
+    ).resolves.toMatchSnapshot();
+  });
+});
+```
+
+Method `run()` will transform all components that are represented by string to corresponding instances, check whether map is locally present based on super.json, loads recording or starts recording, runs perform for given usecase, ends the recording and returns **result** or **error** value from perform.
+
+You can then use this return value to test your capabilities (We recommend you to use jest [snapshot testing](https://jestjs.io/docs/snapshot-testing) as seen in example above).
+
+### Recording
```

Comment:
Nitpicky: Maybe add some general information about two (life/recorded) possible ways of using this repo to the top of the README.